applications/StructuralMechanicsApplication/custom_constitutive/constitutive_laws_integrators/generic_constitutive_law_integrator_kinematic_plasticity.h

```diff
@@ -179,8 +179,12 @@ class GenericConstitutiveLawIntegratorKinematicPlasticity
         const Vector& rPreviousStressVector
         )
     {
+        // Material properties
+        const Properties& r_material_properties = rValues.GetMaterialProperties();
+
+        // Defining some variables
         bool is_converged = false;
-        IndexType iteration = 0, max_iter = 100;
+        IndexType iteration = 0, max_iter = r_material_properties.Has(MAX_NUMBER_NL_CL_ITERATIONS) ? r_material_properties.GetValue(MAX_NUMBER_NL_CL_ITERATIONS) : 100;
```

Comment:
good feature

Reply:
yep, I was tired of hardcoding one to debug